### PR TITLE
Db layer improvements

### PR DESCRIPTION
**Pull request by i041934**
# Pull Request Template

## Prerequisites

https://jtrack.wdf.sap.corp/browse/SAPCPCFS-14735

## Motivation

Some of the DB read request are wasteful and not optimized.

e.g. for global vs. subaccount scoped brokers - global brokers are identified by retrieving all brokers and looping to find out brokers w/o subaccount ID

## Approach

An additional method for querying the DB was introduced: a named query

## Pull Request status

* [x] Initial implementation
* [ ] Refactoring
* [x ] Unit tests
* [ ] Integration tests

Feel free to construct the checklist with whatever items seem most reasonable for your change. You could disassemble the Implementation part to even smaller separate checklist items if you're working on something big for example. But do make the effort to provide a checklist of some sort so that the core team as well as the community can have an idea about the progress of your Pull Request.

### CODE REVIEW

File: storage/named_query.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+const (
+	QueryByMissingLabel NamedQuery = iota
+	QueryByExistingLabel
+)
+
+var namedQueries =  map[NamedQuery]string {
+
+	QueryByMissingLabel : `
```
Comment (pavelmaliy) [inline storage/named_query.go:14]: why do you need column alias here?

File: storage/postgres/query_builder_test.go [hunk 2]
```diff
@@ -806,4 +824,101 @@ WHERE visibilities.id = t.id RETURNING *;`)))
...
 	})
+
+	Describe("Query", func() {
+		Context("when query by missing label with params ", func() {
+			It("builds a valid query", func() {
+				params := map[string]interface{}{
+					"key": "subaccount_id"}
+
+				_, err := qb.NewQuery(entity).Query(ctx,storage.QueryByMissingLabel,params)
+				Expect(err).To(HaveOccurred())
```
Comment (pavelmaliy) [inline storage/postgres/query_builder_test.go:835]: no need to validate not real errors, it's just stubbing issue

File: storage/postgres/query_builder_test.go [hunk 3]
```diff
@@ -806,4 +824,101 @@ WHERE visibilities.id = t.id RETURNING *;`)))
...
+				WHERE key=:key
+				AND visibilities.id = visibility_labels.visibility_id)`))
+				//Expect(err.Error()).To(ContainSubstring("query builder requires the entity to have associated label entity"))
+			})
+		})
+
+		Context("when query by missing label with nil params", func() {
+			It("builds simple query for entity and its labels", func() {
+				_, err := qb.NewQuery(entity).Query(ctx,storage.QueryByMissingLabel,nil)
+				Expect(err).Should(HaveOccurred())
```
Comment (pavelmaliy) [inline storage/postgres/query_builder_test.go:858]: no need err validation

File: storage/postgres/query_builder_test.go [hunk 4]
```diff
@@ -806,4 +824,101 @@ WHERE visibilities.id = t.id RETURNING *;`)))
...
+			})
+		})
+
+		Context("when query by missing label with empty params", func() {
+			It("builds simple query for entity and its labels", func() {
+				params := map[string]interface{}{}
+
+				_, err := qb.NewQuery(entity).Query(ctx,storage.QueryByMissingLabel,params)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("could not prepare statement"))
```
Comment (pavelmaliy) [inline storage/postgres/query_builder_test.go:870]: you test nothing here

File: storage/postgres/query_builder_test.go [hunk 5]
```diff
@@ -806,4 +824,101 @@ WHERE visibilities.id = t.id RETURNING *;`)))
...
+		ON visibilities.id = visibility_labels.visibility_id
+	WHERE EXISTS
+	(SELECT ID FROM visibility_labels 
+				WHERE key=:key
+				AND visibilities.id = visibility_labels.visibility_id)`))
+				//Expect(err.Error()).To(ContainSubstring("query builder requires the entity to have associated label entity"))
+			})
+		})
+
+		Context("when query by existing label with nil params", func() {
```
Comment (pavelmaliy) [inline storage/postgres/query_builder_test.go:903]: you test nothing here

File: storage/postgres/query_builder_test.go [hunk 6]
```diff
@@ -806,4 +824,101 @@ WHERE visibilities.id = t.id RETURNING *;`)))
...
+			It("builds simple query for entity and its labels", func() {
+				_, err := qb.NewQuery(entity).Query(ctx,storage.QueryByExistingLabel,nil)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("could not prepare statement"))
+			})
+		})
+
+		Context("when query by existing label with empty params", func() {
+			It("builds simple query for entity and its labels", func() {
```
Comment (pavelmaliy) [inline storage/postgres/query_builder_test.go:912]: same

File: storage/named_query.go (hunk 1 above)
Comment (i041934) [reply storage/named_query.go:14]: the query fails if the alias are removed

**Pull request closed without merging**